glean-core/rlb/src/net/mod.rs

```diff
@@ -59,6 +60,8 @@ struct Inner {
     uploader: Box<dyn PingUploader + 'static>,
     thread_running: AtomicState,
     handle: Mutex<Option<JoinHandle<()>>>,
+    rx: Receiver<()>,
```

Comment:
We shouldn't need to store this in the struct, do we?
Should be good enough to be moved into the thread.

Reply:
Meaning... `tx` becomes `Option<Sender<()>>` which is `None` when there's no thread? Synchronizing modifications to this via `handle`, I guess.

Wait. While I'm looking at this data... why is `inner` not wrapped in a `Mutex`?

Reply:
Because we never need to change `inner`, only some parts of it (which are either atomic or a Mutex itself)

Reply:
Trying to do this gets a little complex since `inner` isn't `mut` so giving it its `tx` late is a big ol' nope.

Mind if I land this as-is?